scrape/target.go

```diff
@@ -125,13 +125,15 @@ func (t *Target) hash() uint64 {
 }
 
 // offset returns the time until the next scrape cycle for the target.
-func (t *Target) offset(interval time.Duration) time.Duration {
+// It includes the global server jitter for scrapes of HA setup to be spread across the time interval.
+func (t *Target) offset(interval, jitter time.Duration) time.Duration {
 	now := time.Now().UnixNano()
 
+	// base is a pinned to a absolute time, no matter how often offset is called.
 	var (
 		base   = int64(interval) - now%int64(interval)
 		offset = t.hash() % uint64(interval)
-		next   = base + int64(offset)
+		next   = base + int64(offset) + int64(jitter)
```

Comment:
This could delay things by up to a minute, which is undesirable. You should augment the above hash, rather than adding a new time period to worry about.

Reply:
it could be more than the interval itself, see [below](https://github.com/prometheus/prometheus/pull/5181/files#diff-71e6b8df5f2eeb6a786cb5cacaa4187dR139) :

```
	if next > int64(interval) {
		next -= int64(interval)
	}
```

so it won't add the more diff than offset from hash `t.hash() % uint64(interval)`, it just adds an entropy throughout the HA setup.

Reply:
If the interval is 1ms, then it can delay things by a minute.

Reply:
fair, will change, thanks for catch.